analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -809,19 +809,19 @@ public void TypeOf_SetsNotNull()
         }
 
         [DataTestMethod]
-        [DataRow("bool", "true", "False")]
-        [DataRow("bool", "false", "True")]
-        [DataRow("bool?", "default", "Null")]
-        [DataRow("bool?", "null", "Null")]
-        public void Unary_Not_SupportsBoolAndNull(string type, string defaultValue, string expectedConstraints)
+        [DataRow("bool", "true", new[] { ConstraintKind.BoolFalse })]
+        [DataRow("bool", "false", new[] { ConstraintKind.BoolTrue })]
+        [DataRow("bool?", "default", new[] { ConstraintKind.ObjectNull })]
+        [DataRow("bool?", "null", new[] { ConstraintKind.ObjectNull })]
+        public void Unary_Not_SupportsBoolAndNull(string type, string defaultValue, ConstraintKind[] expectedConstraints)
```

Comment:
Same here, we expect exactly one. The assertion should reflect that.
```suggestion
        public void Unary_Not_SupportsBoolAndNull(string type, string defaultValue, ConstraintKind expectedConstraints)
```